src/components/mobile/MobileLayout.js

```diff
@@ -39,21 +40,29 @@ const MobileLayout = () => {
       <TabPanels>
         <TopBarSwitch />
         <Switch>
-          <Route path="/entry/new/details">
+          <Route path={['/map/entry/new/details']}>
             <LocationForm />
           </Route>
-          <Route path={['/map', '/list', '/settings', '/entry/new']}>
-            {tabPanels}
-          </Route>
-          <Route path="/entry/:id">
+          <Route path="/list/entry/:id">
             <EntryTabs />
           </Route>
+          <Route path={['/map', '/list', '/settings', '/map/entry/new']}>
+            <Switch>
+              <Route path={'/map/entry/new'} />
+              <Route path="/map/entry/:id">
+                <Drawer>
+                  <EntryTabs />
+                </Drawer>
+              </Route>
+            </Switch>
+            {tabPanels}
+          </Route>
         </Switch>
       </TabPanels>
       <Switch>
-        <Route path="/entry/new/details"></Route>
+        <Route path={['/map/entry/new/details']}></Route>
         <Route>
-          <TabList>{tabList}</TabList>
+          <TabList style={{ zIndex: 999 }}>{tabList}</TabList>
```

Comment:
I would add a TODO here to 1) use the z-index enum and 2) override this in `ui/PageTabs` with a selector.